### DISCUSSION TITLE
Is the Istio CNI plugin supported by deployKF ?

### DISCUSSION THREAD

**Discussion by thib12** [deployKF/deployKF#77] [Questions & Answers] [answered]
The Istio Container Network Interface ([CNI](https://istio.io/latest/docs/setup/additional-setup/cni/)) plugin is required on our side as we do no want to give elevated Kubernetes RBAC permissions.

Is this plugin compatible with deployKF ?
I see that it is directly available in Kubeflow's manifests (see https://github.com/kubeflow/manifests/tree/master/common/istio-cni-1-17).

Thanks !

**Comment by thesuperzapper** [comment, marked as answer]
@thib12 I am interested to know, are you willing to test it out in your environment? 

You can follow the ["bring your own istio"](https://www.deploykf.org/guides/dependencies/istio/#can-i-use-my-existing-istio) guide and just install the CNI rather than core Istio.

---

Also, as a side note, I want to highlight that using the CNI plugin still means you need to create a DemonSet (a pod that runs with root privileges on literally every node), but I do agree that it's slightly better than giving every pod special network permissions.

**Reply by thib12** [reply]
Yes, it seems to work. Thanks !

**Comment by berglh** [comment]
@thib12 Did you disable the sidecar injection or just leave it as default in deploykF? I am also running the CNI plugin, but I noticed that the deployKF components still get the sidecar injected from the namespace configuration.
